StudySwift/StudySwift/Landing Screen/Profile Screen/ProfileScreenView.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  ProfileScreenView.swift
+//  StudySwift
+//
+//  Created by Myat Thu Ko on 5/1/23.
+//
+
+import SwiftUI
+
+struct ProfileScreenView: View {
+
+    @ObservedObject var viewModel: ProfileScreenViewModel
+    @State private var errorMessage: String = ""
+    @State private var isLogout: Bool = false
+    @State private var isPasswordValid: Bool = true
+
+    var body: some View {
+        NavigationView {
+            VStack {
+                Image("mainLogo")
+                    .resizable()
+                    .scaledToFit()
+                    .padding()
+
+                Spacer()
+
+                VStack(alignment: .leading, spacing: 25) {
+                    TextField("Email", text: $viewModel.email)
+                        .textFieldStyle(RoundedTextFieldStyle())
+                        .textInputAutocapitalization(.never)
+
+                    SecureField("Old Password", text: $viewModel.oldPassword)
+                        .textFieldStyle(RoundedTextFieldStyle())
+
+                    SecureField("New Password", text: $viewModel.newPassword)
+                        .textFieldStyle(RoundedTextFieldStyle())
+
+                    SecureField("Confirm Password", text: $viewModel.confirmPassword)
+                        .textFieldStyle(RoundedTextFieldStyle())
+
+                    if !isPasswordValid {
+                        Text(errorMessage)
+                            .fixedSize(horizontal: false, vertical: true)
+                            .padding(.horizontal)
+                            .foregroundColor(.red)
+                    }
+                }
+                .padding(.horizontal, 16)
+
+                Spacer()
+
+                Button {
+                    if !viewModel.isValidPassword() {
+                        isPasswordValid = false
+                        errorMessage = "Password must be at least 6 characters long and contain at least one uppercase letter, one lowercase letter, and one special character."
+                    } else if !viewModel.isSamePassword() {
```

Comment:
I think this should be `} else if viewModel.isSamePassword() {` since `!viewModel.isSamePassword()` will force us into this condition if the old password & new password doesn't match